mypy/checker.py

```diff
@@ -1874,7 +1874,8 @@ def check_return_stmt(self, s: ReturnStmt) -> None:
                 if isinstance(typ, AnyType):
                     # (Unless you asked to be warned in that case, and the
                     # function is not declared to return Any)
-                    if not isinstance(return_type, AnyType) and self.options.warn_return_any:
+                    if (not is_proper_subtype(AnyType(), return_type) and
+                            self.options.warn_return_any):
```

Comment:
I would have tested for the option first, an attribute access is cheaper than a call. It's a reflexive optimization that often also leads to clearer code (often it's a wash).